sdk/storage/Azure.Storage.Blobs/tests/PageBlobClientOpenWriteTests.cs

```diff
@@ -0,0 +1,247 @@
+
+            // Act
+            await TestHelper.AssertExpectedExceptionAsync<ArgumentException>(
+                blob.OpenWriteAsync(
+                    overwrite: true,
+                    position: 0),
+                e => Assert.AreEqual("options.Size must be set if overwrite is set to true", e.Message));
+        }
+
+        [RecordedTest]
+        public async Task OpenWriteAsync_NewBlobNoSize()
+        {
+            // Arrange
+            await using IDisposingContainer<BlobContainerClient> disposingContainer = await GetDisposingContainerAsync();
+            PageBlobClient blob = GetResourceClient(disposingContainer.Container);
+
+            // Act
+            await TestHelper.AssertExpectedExceptionAsync<ArgumentException>(
+                blob.OpenWriteAsync(
+                    overwrite: false,
+                    position: 0),
+                e => Assert.AreEqual("options.Size must be set if the Page Blob is being created for the first time", e.Message));
+        }
+
+        [RecordedTest]
+        public async Task OpenWriteAsync_Position()
+        {
+            // Arrange
+            await using IDisposingContainer<BlobContainerClient> disposingContainer = await GetDisposingContainerAsync();
+            PageBlobClient blob = GetResourceClient(disposingContainer.Container);
+            await blob.CreateAsync(Constants.KB);
+
+            byte[] data0 = GetRandomBuffer(512);
+            byte[] data1 = GetRandomBuffer(512);
+            using Stream dataStream0 = new MemoryStream(data0);
+            using Stream dataStream1 = new MemoryStream(data1);
+            byte[] expectedData = new byte[Constants.KB];
+            Array.Copy(data0, expectedData, 512);
+            Array.Copy(data1, 0, expectedData, 512, 512);
+
+            // Act
+            Stream openWriteStream = await blob.OpenWriteAsync(
+                overwrite: false,
+                position: 0);
+
+            Assert.AreEqual(0, openWriteStream.Position);
+
+            await dataStream0.CopyToAsync(openWriteStream);
+
+            Assert.AreEqual(512, openWriteStream.Position);
+
+            await dataStream1.CopyToAsync(openWriteStream);
+
+            Assert.AreEqual(1024, openWriteStream.Position);
+
+            await openWriteStream.FlushAsync();
+
+            Assert.AreEqual(1024, openWriteStream.Position);
+
+            Response<BlobDownloadInfo> result = await blob.DownloadAsync();
+            MemoryStream dataResult = new MemoryStream();
+            await result.Value.Content.CopyToAsync(dataResult);
+            Assert.AreEqual(expectedData.Length, dataResult.Length);
+            TestHelper.AssertSequenceEqual(expectedData, dataResult.ToArray());
+        }
+
+        /// <summary>
+        /// Override test for page-alligned results, leaving more arbitrary numbers to still run for other services.
+        /// </summary>
+        /// <returns></returns>
+        [RecordedTest]
```

Comment:
Why are there more tests in `PageBlobClientOpenWriteTests` than the other `*ClientOpenWriteTests`?

Reply:
PageBlobClient had more open write tests that appeared to tackle page-blob-specific functionality. Made more sense to add those tests to the pageblob class extension than to add them to the base test class only to get overwritten in every other subclass.